src/midiEditorCommands.cpp

```diff
@@ -455,6 +484,9 @@ void cmdMidiMovePitchCursor(Command* command) {
 	int chan = MIDIEditor_GetSetting_int(editor, "default_note_chan");
 	int vel = MIDIEditor_GetSetting_int(editor, "default_note_vel");
 	previewNotes(take, {{chan, pitch, vel}});
```

Comment:
Perhaps we could use the configured insertion length as the preview length for the pitch cursor? We don't have to do that in this PR, though.

Reply:
I looked into that, but the configured insertion length is in midi ticks (ppq), and we can only convert a start and end to project time values, converting a length fails. This is what I tried:

* Converting the length from ppq to time directly
* Getting start as the cursor position, convert it to ppq, add the length ppq and convert the end back to project time

All resulted in a too short preview, probably around USER_TIMER_MINIMUM (0x0000000A) in length. If you have alternative suggestions, I'd be glad to know them.

Reply:
I can understand why converting the raw length might not work, but the second surprises me. I think we'd need to get a look at the raw values we're getting back from each function to know what's going on there.

Is the insertion length from MIDIEditor_GetSetting_int even correct?

Failing that, I wonder whether MIDI_GetPPQPosFromProjTime expects a time relative to the take or something like that? That would make the two functions asymmetric, but this wouldn't be the first time I've been confused by REAPER's lack of documentation/consistency on the time base for various functions.

Alternatively, we could convert to qn time and then back, which is absolute to the project instead of per take. We'd need to calculate the length in quarter notes ourselves. Instead of using MIDIEditor_GetSetting_int, we could use GetToggleCommandState with the various insertion length action ids, converting 1/4 to 1 qn, 1/2 to 2 qn, etc.

Reply:
I'm afraid that whatever we do that relates to project time will at least always fail if it is the first note we're inserting, as the project is empty at that point.